pandas/_libs/lib.pyx

```diff
@@ -716,6 +716,14 @@ cpdef ndarray[object] ensure_string_array(
         Py_ssize_t i = 0, n = len(arr)
 
     if hasattr(arr, "to_numpy"):
```

Comment:
we dont' have access to the ExtnsionTypes here right? so this is just as easy

Reply:
right